source/common/conn_pool/conn_pool_base.cc

```diff
@@ -363,6 +387,11 @@ void ConnPoolImplBase::closeIdleConnectionsForDrainingPool() {
     for (auto& client : connecting_clients_) {
       to_close.push_back(client.get());
     }
+    for (auto& client : early_data_clients_) {
```

Comment:
Just to confirm... There are a lot of places in this class where it looks at the various client related members. Have you checked all such sites to make sure that we added early_data_clients_ to all the right places? (I suspect you have, just want to confirm)

Reply:
Yes, I audited all the call sites of *clients_ in this base class.